imsim/diffraction.py

```diff
@@ -0,0 +1,166 @@
+"""Routines to implement diffraction due to the LSST spider.
+
+The core idea is taken from https://ntrs.nasa.gov/citations/19990094899
+For more details, see doc/diffraction.md (Statistical Aproach)
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from collections.abc import Callable
+
+import numpy as np
+
+
+@dataclass
+class Geometry:
+    """Defining a 2D geometry consisting of circles and lines with a thickness.
+
+    Parameters
+    ----------
+    thick_lines : numpy array of shape (n,4) where each row has the form
+        [nx, ny, d, thickness]. (nx, ny) is the normal to the line and d is
+        such that d*(nx, ny) lies on the line.
+    circles : numpy array of shape (n,3) where each row has the form
+        [x, y, r]. (x,y) is the center of the circle, and r the radius.
+    """
+
+    thick_lines: np.ndarray
+    circles: np.ndarray
+
+
+LSST_SPIDER_GEOMETRY = Geometry(
+    thick_lines=np.array(
+        [
+            [1 / np.sqrt(2.0), 1 / np.sqrt(2.0), -0.4, 0.025],
+            [-1 / np.sqrt(2.0), 1 / np.sqrt(2.0), -0.4, 0.025],
+            [1 / np.sqrt(2.0), 1 / np.sqrt(2.0), 0.4, 0.025],
+            [-1 / np.sqrt(2.0), 1 / np.sqrt(2.0), 0.4, 0.025],
+        ]
+    ),
+    circles=np.array([[0.0, 0.0, 2.3905], [0.0, 0.0, 4.35], [0.0, 0.0, 2.4005]]),
```

Comment:
I think we just want radii of 4.18 and 2.558 (the inner/outer radius of M1) for now.  We'll need to think harder about this to get off-axis right, but we can postpone that for now.